plugins/extractors/bigquery/bigquery.go

```diff
@@ -301,38 +329,50 @@ func (e *Extractor) extractTable(ctx context.Context, ds *bigquery.Dataset, emit
 			continue
 		}
 
+		var wg sync.WaitGroup
+		wg.Add(len(tables))
 		for _, table := range tables {
-			if IsExcludedTable(ds.DatasetID, table.TableID, e.config.Exclude.Tables) {
-				e.excludedTableCtr.Add(ctx, 1, metric.WithAttributes(
-					attribute.String("bq.project_id", e.config.ProjectID),
-					attribute.String("bq.dataset_id", ds.DatasetID),
-				))
-				e.logger.Debug("excluding table from bigquery extract", "dataset_id", ds.DatasetID, "table_id", table.TableID)
-				continue
-			}
-
-			tableFQN := table.FullyQualifiedName()
-
-			e.logger.Debug("extracting table", "table", tableFQN)
-			tmd, err := e.fetchTableMetadata(ctx, table)
-			if err != nil {
-				e.logger.Error("failed to fetch table metadata", "err", err, "table", tableFQN)
-				continue
-			}
-
-			asset, err := e.buildAsset(ctx, table, tmd)
-			if err != nil {
-				e.logger.Error("failed to build asset", "err", err, "table", tableFQN)
-				continue
-			}
-
-			emit(models.NewRecord(asset))
+			go e.processTable(ctx, ds, table, emit, &wg)
 		}
 
+		wg.Wait()
+
 		if !hasNext {
 			break
 		}
 	}
+
+	return nil
+}
+
+func (e *Extractor) processTable(ctx context.Context, ds *bigquery.Dataset, table *bigquery.Table, emit plugins.Emit, wg *sync.WaitGroup) {
+	defer wg.Done()
```

Comment:
should be consistent with the processDatasetWorker approach, can move this outside the processTable function